### DISCUSSION TITLE
Is there a way to open MiniFiles focused on the current buffers file, but with the anchor as the current working directory?

### DISCUSSION THREAD

**Discussion by IndianBoy42** [nvim-mini/mini.nvim#872] [Q&A] [answered]
`MiniFiles.open(vim.api.nvim_buf_get_name(0))` focuses on the file but the anchor is the immediate parent of the file. i'd like it to be rooted at the cwd the same way as `MiniFiles.open()` , just opening however many levels deep of nested directory as needed.

**Comment by echasnovski** [comment, marked as answer]
Sorry, but I don't think there is a reasonable way to support that. The best compromise for this is the suggested `MiniFiles.open(vim.api.nvim_buf_get_name(0))` approach but with `@` mapping to reveal the current working directory.

**Reply by IndianBoy42** [reply]
wouldn't it be possible to add a flag to the third argument to `MiniFiles.open` that when set you call `reveal_cwd` when the explorer opens? I tried to recreate it using the `MiniFilesBufferCreate` event but I couldn't figure out a neat way to get the opts, except a separate global variable. 

Adding this code works
```lua
MiniFiles.open = function(path, use_latest, opts)
   -- Existing Code
   -- ....

	if explorer.opts.options.reveal_cwd_on_open then
		MiniFiles.reveal_cwd()
	end
end

```

**Reply by IndianBoy42** [reply]
Oh I hadn't updated in a while `MiniFilesExplorerOpen` would work except I'm not sure how to get the options of the current explorer session

**Reply by echasnovski** [reply]
Yes, `MiniFilesExplorerOpen` was added as a reaction to #817, as a way for external tools/plugins to act on user opening explorer.

For this particular case of "automatically reveal cwd for explorer opened at file", I'd suggest writing own wrapper. So something like this:

```lua
local open_at_buf_file = function()
  MiniFiles.open(vim.api.nvim_buf_get_name(0))
  MiniFiles.reveal_cwd()
end
vim.keymap.set('n', '<Leader>ef', open_at_buf_file)
```
